java/src/jmri/jmrit/logixng/actions/swing/ActionSignalHeadSwing.java

```diff
@@ -252,79 +267,94 @@ protected void createPanel(@CheckForNull Base object, @Nonnull JPanel buttonPane
             _signalHeadAppearanceReferenceTextField.setText(action.getAppearanceReference());
             _signalHeadAppearanceLocalVariableTextField.setText(action.getAppearanceLocalVariable());
             _signalHeadAppearanceFormulaTextField.setText(action.getAppearanceFormula());
-            
-            SignalHead sh = null;
-            if ((_tabbedPaneSignalHead.getSelectedComponent() == _panelSignalHeadDirect)
-                    && (action.getSignalHead() != null)) {
-                sh = action.getSignalHead().getBean();
-            } else if (action.getExampleSignalHead() != null) {
-                sh = action.getExampleSignalHead().getBean();
-            }
-            
-            if (sh != null) {
-                int[] states = sh.getValidStates();
-                for (int s : states) {
-                    SignalHeadAppearance sha = new SignalHeadAppearance();
-                    sha._state = s;
-                    sha._name = sh.getAppearanceName(s);
-                    _signalHeadAppearanceComboBox.addItem(sha);
-                    if (action.getAppearance() == s) _signalHeadAppearanceComboBox.setSelectedItem(sha);
-                }
-                JComboBoxUtil.setupComboBoxMaxRows(_signalHeadAppearanceComboBox);
-            }
+
+            setAppearanceComboBox(action);
```

Comment:
I suggest you change this line to:
`jmri.util.ThreadingUtil.runOnGUIEventually(() -> { setAppearanceComboBox(action); });`